src/main/java/com/dougnoel/sentinel/webdrivers/WebDriverFactory.java

```diff
@@ -92,32 +94,57 @@ public static WebDriver instantiateWebDriver() {
+        	
+        	//use full browser version from grid in yml or command line e.g: 88.0.4324.96
+        	driver = instantiateRemoteWebDriver(browser, browserVersion, gridUrl);
+        }else {
+            // Initialize the driver object based on the browser and operating system (OS).
+            // Throw an error if the value isn't found.   	
+        	switch (browser) {
+            case "chrome":
+            	String headless = ConfigurationManager.getOptionalProperty("headless");
+            	if (headless == null || headless.equalsIgnoreCase("false"))
+            		driver = ChromeDriverFactory.createChromeDriver();
+            	else
+            		driver = ChromeDriverFactory.createHeadlessChromeDriver();
+                break;
+            case "firefox":
+            	driver = FirefoxDriverFactory.createFirefoxDriver();
+                break;
+            case "internetexplorer":
+            	driver = createInternetExplorerDriver();
+                break;
+            case "safari":
+            	driver = createSafariDriver();
+                break;
+            default:
+                throw new WebDriverException(SentinelStringUtils.format("Invalid browser type '{}' passed to WebDriverFactory. Could not resolve the reference. Check your spelling. Refer to the Javadoc for valid options.", browser));
+            }
+        }
 
-        String browser = ConfigurationManager.getBrowserName();
+        
 
-        // Initialize the driver object based on the browser and operating system (OS).
-        // Throw an error if the value isn't found.   	
-    	switch (browser) {
-        case "chrome":
-        	String headless = ConfigurationManager.getOptionalProperty("headless");
-        	if (headless == null || headless.equalsIgnoreCase("false"))
-        		driver = ChromeDriverFactory.createChromeDriver();
-        	else
-        		driver = ChromeDriverFactory.createHeadlessChromeDriver();
-            break;
-        case "firefox":
-        	driver = FirefoxDriverFactory.createFirefoxDriver();
-            break;
-        case "internetexplorer":
-        	driver = createInternetExplorerDriver();
-            break;
-        case "safari":
-        	driver = createSafariDriver();
-            break;
-        default:
-            throw new WebDriverException(SentinelStringUtils.format("Invalid browser type '{}' passed to WebDriverFactory. Could not resolve the reference. Check your spelling. Refer to the Javadoc for valid options.", browser));
-        }
+        return driver;
+    }
+    
+    private static WebDriver instantiateRemoteWebDriver(String browser, String browserVersion, String gridUrl) {
 
+    	DesiredCapabilities capability = new DesiredCapabilities();
+    	capability.setCapability("browserName", browser);
+    	capability.setCapability("browserVersion", browserVersion);
+    	
+        try {
+            driver = new RemoteWebDriver(new URL(gridUrl), capability);
+        } catch (Exception e) {
```

Comment:
We should never catch and dump an error. And we should always be as specific as possible with error handling. What exceptions are we expecting here? Your IDE will tell you if you remove the try/catch block. Catch it and throw a well-formatted WebDriverException() to give the end user as much possible information.

Reply:
If we don't use try catch then you have to throw malformedurlexception, should I use that and remove try catch ?

Reply:
Catch the MalformedURLException and wrap it in one of ours and throw it. Take a look at lines 25-29 of the SaucelabsDriverFactory to see how I handled it. Not that I'm throwing a sentinel Exception there even though it looks the same.

And yes, we absolutely should be throwing exceptions like this because without a driver we cannot continue.